tests/trainer/test_trainer_error_handling.py

```diff
@@ -0,0 +1,49 @@
+# Copyright The PyTorch Lightning team.
```

Comment:
would it not be better to move the tests to `tests/trainer/trainer.py`?

Reply:
I believe it was moved out to a new file due to `test_trainer.py`'s size

Reply:
yes that is correct, as per @ananthsub 's suggestion. I dont have a strong preference either way

Reply:
@awaelchli @carmocca do you prefer this test case in the main `test_trainer.py` ? my motivation for moving it out was that the main test file had gotten so large that it was difficult to check for coverage of specific functionality

Reply:
no strong preference but the pattern we were trying to adapt is to put tests for `a/b/c.py` into `tests/a/b/c.py` so that we know what is tested where.